aspnetcore/blazor/tutorials/movie-database-app/part-2.md

```diff
@@ -0,0 +1,500 @@
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace BlazorWebAppMovies.Models;
+
+public class Movie
+{
+    public int Id { get; set; }
+
+    public string? Title { get; set; }
+
+    public DateOnly ReleaseDate { get; set; }
+
+    public string? Genre { get; set; }
+
+    [DataType(DataType.Currency)]
+    [Column(TypeName = "decimal(18, 2)")]
+    public decimal Price { get; set; }
+}
+```
+
+The `Movie` class contains:
+
+* A record identifier property (`Id`), which is required by EF Core and the database to track records. In the database, the `Id` property is the database record's primary key.
+* Other properties that describe aspects of a movie:
+  * Title (`Title`)
+  * Release date (`ReleaseDate`)
+  * Genre (`Genre`)
+  * Price (`Price`)
+
+The question mark on a `string` type indicates that the property is nullable (it can hold a `null` value).
+
+The EF Core database provider selects data types based on the .NET types of the model's properties. The provider also takes into account other metadata provided by <xref:System.ComponentModel.DataAnnotations>, which are a set of attribute classes placed above a model's property with the following format, where the `{ANNOTATION}` placeholder is the annotation name. You can place multiple annotations on the same line separated by commas inside the brackets, or you can place multiple annotations on separate lines, which is the approach adopted by this tutorial series. 
+
+```csharp
+[{ANNOTATION}]
+```
+
+The `Price` property in the `Movie` class file has two data annotations:
+
+```csharp
+[DataType(DataType.Currency)]
+[Column(TypeName = "decimal(18, 2)")]
+public decimal Price { get; set; }
+```
+
+These annotations specify:
+
+* That the property holds a currency data type.
+* The database column is a decimal of 18 digits with two decimal places.
+
+More information on data annotations, including adding data annotations for validation, is covered in a later part of the tutorial series.
+
+:::zone pivot="vs"
+
+Select **Build** > **Build Solution** from the menu bar or press <kbd>F6</kbd> on the keyboard. Confirm in the **Output** panel that the build succeeded.
+
+:::zone-end
+
+:::zone pivot="vsc"
+
+## Add Nuget packages and tools
+
+To add the required NuGet packages and tools, execute the following .NET CLI commands in the **Terminal** (**Terminal** menu > **New Terminal**).
+
+Paste all of the following commands at the prompt (`>`) of the **Terminal**. When you paste multiple commands, a warning appears stating that multiple commands will execute. Dismiss the warning and proceed with the paste operation.
+
+When you paste multiple commands, all of the commands execute except the last one. The last command doesn't execute until you press <kbd>Enter</kbd> on the keyboard.
+
+```dotnetcli
+dotnet tool install --global dotnet-aspnet-codegenerator
+dotnet tool install --global dotnet-ef
+dotnet add package Microsoft.EntityFrameworkCore.Design
```

Comment:
I think Microsoft.EntityFrameworkCore.Design is a transitive dependency of Microsoft.EntityFrameworkCore.Tools, so installing it separately shouldn't be needed.